lib/api/src/entities/engine/mod.rs

```diff
@@ -0,0 +1,126 @@
+//! Defines the [`Engine`] type, the [`EngineLike`] trait for implementors and useful
+//! traits and data types to interact with them.
+
+use bytes::Bytes;
+use std::{path::Path, sync::Arc};
+use wasmer_types::DeserializeError;
+
+#[cfg(feature = "sys")]
+use wasmer_compiler::Artifact;
+
+use crate::{IntoBytes, Store};
+
+/// Create temporary handles to engines.
+mod engine_ref;
+
+/// The actual (private) definition of the engines.
+mod inner;
+pub(crate) use inner::RuntimeEngine;
+
+pub use engine_ref::*;
+
+static ENGINE_ID_COUNTER: std::sync::atomic::AtomicU64 = std::sync::atomic::AtomicU64::new(0);
+
+pub(crate) fn atomic_next_engine_id() -> u64 {
```

Comment:
The ID should not be a plain u64.

We want to have the ability to change it.
It should be an opaque type like

```
#[derive(Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Debug)]
struct EngineId(u64);
```